src/main/groovy/org/shipkit/internal/notes/generator/DefaultReleaseNotesGenerator.java

```diff
@@ -48,7 +48,7 @@
         Collection<ReleasedVersion> versions = releasedVersionsProvider.getReleasedVersions(headVersion, new Date(), targetVersions, tagPrefix);
 
         for (ReleasedVersion v : versions) {
-            if (v.getPreviousRev() == null) {
+            if (versions.size() > 1 && v.getPreviousRev() == null) {
```

Comment:
The complexity around handling versions is getting a bit out of hand. However, I don't have any improvement suggestion. So, the only thing I can do is to approve this PR :)